git-importer.py

```diff
@@ -917,6 +927,24 @@ def import_all_revisions(self):
         for revision in revisions:
             self.import_revision(revision)
 
+    def import_new_revision_with_request(self, revision, request):
+        """Create a new branch as a result of a merge"""
+
+        submitted_revision = self.history.find_revision(
+            request.source, request.revisionid, revision.time
+        )
+        if not submitted_revision:
+            logging.warning(f"Request {request} does not connect to a known revision")
+            return False
+        assert submitted_revision.commit is not None
```

Comment:
this assertion hits 60 packages now. 

```
(venv) importer@git-importer:~/logs> cat FAIL.premake4
CACHE_DIR /home/importer/.cache/openSUSE-release-tools/request/main
INFO:root:Importing [Rev openSUSE:Factory/1 Md5 9b55eae0a1f723d7d9f7411f87b271cc 2011-04-29 10:50:37 saschpe 68121] to factory
Traceback (most recent call last):
  File "/home/importer/openSUSE-release-tools/git-importer.py", line 1192, in <module>
    main()
  File "/home/importer/openSUSE-release-tools/git-importer.py", line 1188, in main
    importer.import_all_revisions()
  File "/home/importer/openSUSE-release-tools/git-importer.py", line 940, in import_all_revisions
    self.import_revision(revision)
  File "/home/importer/openSUSE-release-tools/git-importer.py", line 1087, in import_revision
    self.import_new_revision_with_request(revision, request)
  File "/home/importer/openSUSE-release-tools/git-importer.py", line 951, in import_new_revision_with_request
    assert submitted_revision.commit is not None
AssertionError

Reply:
seems that is again an issue with the ordering:

```
DEBUG:root:Selected import order for premake4
DEBUG:root:Rev openSUSE:Factory/1 Md5 9b55eae0a1f723d7d9f7411f87b271cc 2011-04-29 12:50:37 saschpe 68121
DEBUG:root:Rev devel:tools:building/1 Md5 ebc0070db1cced22695e7659707a65b5 2011-04-29 12:50:39 saschpe 68121
```

Reply:
We should build the graph ASAP and drop this mechanism.

Reply:
https://github.com/openSUSE/openSUSE-release-tools/pull/2862